src/util/AdafruitIO_Board.cpp

```diff
@@ -61,6 +64,18 @@ const char* AdafruitIO_Board::type()
     return AdafruitIO_Board::_id;
   }
 
+#elif defined(ARDUINO_ARCH_ESP32)
+
+  char* AdafruitIO_Board::id()
+  {
+    byte mac[6];
+    WiFi.macAddress(mac);
+    for(int i=0; i < 6; i++) {
+      sprintf(&AdafruitIO_Board::_id[i*2],"%02x", mac[i]);
+    }
+    return AdafruitIO_Board::_id;
```

Comment:
i had to use the mac address due to the response from the ESP32 maintainers in [this issue](https://github.com/espressif/arduino-esp32/issues/215).

```
Serial.print("ID: ");
Serial.println(io.boardID());

Serial.print("Type: ");
Serial.println(io.boardType());
```
output:
```
ID: 18fe346a94ec
Type: esp32
```